plugins/inputs/wireless/wireless_mac.go

```diff
@@ -0,0 +1,56 @@
+//go:build darwin
+// +build darwin
+
+package wireless
+
+import (
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+)
+
+// default executable path & flags
+const (
+	OSXCMD = "/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport"
```

Comment:
I'm a little concerned that this is an old, undocumented, and unsupported program. If Apple removes or changes this in the next MacOS release, it would be cause the plugin to stop working.

`airport -I` also only reports on the first wireless device. If users want stats on devices other than the first one, they would be out of luck. I understand many Macs only have one device, but the plugin currently supports multiple devices on Linux so we add support for Mac it would be nice to have feature parity.

It looks like there is a supported api to get these stats in MacOS's Core WLAN framework. https://developer.apple.com/documentation/corewlan. See the CWInterface class.

Golang doesn't have built in support to call into objective c libraries, but it looks like https://github.com/progrium/macdriver might make it possible.

Could you look into calling a public, documented api that supports all interfaces?

Reply:
At this late date in this process, if this is what y'all want, someone else is going to have to do it. I wrote this plug-in 4 years ago, and submitted a pr for it that languished. I was asked late last year to revive it and get it across the finish line. Which I’ve done. That being said, I haven't the time or inclination to start from scratch and write an entirely new plug-in. 

As I’ve said previously in comments to this pr, the `airport` command (and this use of it) has been there for 10+ years. Yes, it could go away. So could almost any api. 

There is no support in macOS for additional wireless devices. I can tell you from first-hand experience that adding a second wireless interface is a heavy lift, and it is not recognized by the macOS apis at all. In the latest version of macOS I have yet to be able to add an additional wireless device at all. 

I’m not sure why it would be important that the values reported match with the Linux plug-in at all. These are entirely different systems so reporting different values is to be expected. But again, if this is deemed critical then we can close this pr and wait for someone to come up with another one for macOS.

Reply:
I should have looked closer at the PR history before suggesting major changes. Thanks for your work and for putting up with the 4 year delay!